### PR TITLE
Ruby Segment changes for Python 2/3:

### PR DESCRIPTION
**Pull request by Blue-Dog-Archolite**
- use "GEM_HOME" in keys
- removal of python2 specific code
- verification of python2 / python3 functionality

### CODE REVIEW

File: powerline_shell/segments/ruby_version.py [hunk 1]
```diff
@@ -10,10 +10,10 @@ def add_to_powerline(self):
             p1 = subprocess.Popen(["ruby", "-v"], stdout=subprocess.PIPE)
             p2 = subprocess.Popen(["sed", "s/ (.*//"], stdin=p1.stdout, stdout=subprocess.PIPE)
             version = p2.communicate()[0].decode("utf-8").rstrip()
-            if os.environ.has_key("GEM_HOME"):
-              gem = os.environ["GEM_HOME"].split("@")
-              if len(gem) > 1:
-                version += " " + gem[1]
+            if "GEM_HOME" in os.environ.keys():
```
Comment (b-ryan) [inline powerline_shell/segments/ruby_version.py]: Can this not be `"GEM_HOME" in os.environ`?

Comment (b-ryan) [reply powerline_shell/segments/ruby_version.py]: Actually, I find this version of the code even nicer:

```python
gem = os.getenv("GEM_HOME", "").split("@")
if len(gem) > 1:
    ...
```

Comment (Blue-Dog-Archolite) [reply powerline_shell/segments/ruby_version.py]: Your very right. I was just patching the issue for the python error. I'll fix it up and make it nice and not just functional.

**Comment by Blue-Dog-Archolite** [comment]
@b-ryan Let me know if that is a better format. I find it easier to read.

**Comment by b-ryan** [comment]
Looks good. Is there ever a case where `GEM_HOME` might not contain a `@` character. If it didn't this would raise an index error.

**Comment by Blue-Dog-Archolite** [comment]
I managed to find one case and that caused the full path of the ruby executable to show and not the gemset.

Code corrected to sort that out.

**Pull request merged**